app/Http/Controllers/User/BookController.php

```diff
@@ -138,65 +142,77 @@ public function store(BookRequest $request)
+                        $isReview = $this->review->checkReview($data);
+                        if ($isReview->count() > 0) {
+                            $flag = false;
+                        }
+                    } else {
+                        $flag = false;
+                    }
 
-            if (Auth::check()) {
-                $userId = Auth::user()->id;
-                $isOwner = in_array($userId, $book->owners->pluck('id')->toArray());
-                $isBooking = in_array($userId, $book->users->pluck('id')->toArray());
-            } else {
-                $isOwner = false;
-                $isBooking = false;
-            }
+                    if (Auth::check()) {
+                        $userId = Auth::user()->id;
+                        $isOwner = in_array($userId, $book->owners->pluck('id')->toArray());
+                        $isBooking = in_array($userId, $book->users->pluck('id')->toArray());
+                    } else {
+                        $isOwner = false;
+                        $isBooking = false;
+                    }
 
 
-            $tmp = false;
-            if (Auth::check()) {
-                $roles = Auth::user()->roles;
-                if ($roles->count() > 0) {
-                    if ($roles->count() > 1) {
-                        foreach ($roles as $role) {
-                            if ($role->name == 'Admin' || $role->name == 'Editor') {
+                    $tmp = false;
+                    if (Auth::check()) {
+                        $roles = Auth::user()->roles;
+                        if ($roles->count() > 0) {
+                            if ($roles->count() > 1) {
+                                foreach ($roles as $role) {
+                                    if ($role->name == 'Admin' || $role->name == 'Editor') {
+                                        $tmp = true;
+                                    }
+                                }
+                            } elseif ($roles[0]->name == 'Admin' || $roles[0]->name == 'Editor') {
                                 $tmp = true;
                             }
                         }
-                    } elseif ($roles[0]->name == 'Admin' || $roles[0]->name == 'Editor') {
-                        $tmp = true;
                     }
-                }
-            }
 
-            $data = [
-                'book',
-                'relatedBooks',
-                'flag',
-                'reviews',
-                'isOwner',
-                'isBooking',
-                'tmp',
-            ];
+                    $data = [
+                        'book',
+                        'relatedBooks',
+                        'flag',
+                        'reviews',
+                        'isOwner',
+                        'isBooking',
+                        'tmp',
+                    ];
+
+                    return view('book.book_detail', compact($data));
+                }
+            } 
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine</br>**Severity**: error</br>**File**: app/Http/Controllers/User/BookController.php</br>**Line**: 208</br>Whitespace found at end of line